### DISCUSSION TITLE
making a console window

### DISCUSSION THREAD

**Discussion by pm100** [emilk/egui#4759] [General]
As an aside to mr milk - what an awesome resource the big demo app is, the fact that I can click on anything and get the source code is a huge help!

I want to make a scrolling console window where users can type commands and get responses (I am building an emulator / debugger combination and want to user to be able to enter debugger commands as opposed to having to navigate menus). I think the best way to do this is
- use a scrollable multiline widget
- then intercept key strokes destined for that window so that , for example, I can change the up arrow key behavior to recall the previous command

I intercept by :
 - checking to see if input.focus() in on my widget
 - then use consume_key for each key that has special meaning for me (up, down, ctrl-r...)

Just want to check that this isnt a bad approach - dont want to spend a week digging myself into a hole

**Comment by qq253498229** [comment]
This is fantastic!

**Reply by pm100** [reply]
now wroking https://github.com/pm100/egui_console
